lib/tftp/src/tftp.erl

```diff
@@ -25,209 +25,127 @@
 -moduledoc """
 Trivial FTP.
 
 Interface module for the `tftp` application.
 
-[](){: #options }
+## Overwiew
+
+This is a complete implementation of the following IETF standards:
+    RFC 1350, The TFTP Protocol (revision 2).
+    RFC 2347, TFTP Option Extension.
+    RFC 2348, TFTP Blocksize Option.
+    RFC 2349, TFTP Timeout Interval and Transfer Size Options.
+
+The only feature that not is implemented in this release is
+the "netascii" transfer mode.
+
+The start/1 function starts a daemon process which, listens for UDP
+packets on a port. When it receives a request for read or write it
+spawns a temporary server process which handles the actual transfer
+of the file. On the client side the read_file/3 and write_file/3
+functions spawns a temporary client process which establishes contact
+with a TFTP daemon and performs the actual transfer of the file.
+
+Most of the options are common for both the client and the server
+side, but some of them differs a little.
+
+## Callbacks
+
+A `tftp` callback module is to be implemented as a `tftp` behavior and export
+the functions listed in the following.
+
+On the server side, the callback interaction starts with a call to `open/5` with
+the registered initial callback state. `open/5` is expected to open the
+(virtual) file. Then either function [`read/1`](`c:read/1`) or
+[`write/2`](`c:write/2`) is invoked repeatedly, once per transferred block. At
+each function call, the state returned from the previous call is obtained. When
+the last block is encountered, function [`read/1`](`c:read/1`) or
+[`write/2`](`c:write/2`) is expected to close the (virtual) file and return its
+last state. Function [`abort/3`](`c:abort/3`) is only used in error situations.
+Function `prepare/5` is not used on the server side.
+
+On the client side, the callback interaction is the same, but it starts and ends
+a bit differently. It starts with a call to `prepare/5` with the same arguments
+as `open/5` takes. `prepare/5` is expected to validate the TFTP options
+suggested by the user and to return the subset of them that it accepts. Then the
+options are sent to the server, which performs the same TFTP option negotiation
+procedure. The options that are accepted by the server are forwarded to function
+`open/5` on the client side. On the client side, function `open/5` must accept
+all option as-is or reject the transfer. Then the callback interaction follows
+the same pattern as described for the server side. When the last block is
+encountered in [`read/1`](`c:read/1`) or [`write/2`](`c:write/2`), the returned
+state is forwarded to the user and returned from `read_file`/3 or
+[`write_file/3`](`write_file/3`).
+
+If a callback (performing the file access in the TFTP server) takes too long
+time (more than the double TFTP time-out), the server aborts the connection and
+sends an error reply to the client. This implies that the server releases
+resources attached to the connection faster than before. The server simply
```

Comment:
before what?

Reply:
I have now idea just part of the original formulation, I will remove this sentence it does not add any value as is.